File: 365/Easy.hs
-- https://www.reddit.com/r/dailyprogrammer/comments/8xbxi9/20180709_challenge_365_easy_uparrow_notation/

up :: Integer -> Integer -> Integer -> Integer
up 1 x y = x^y
up _ _ 0 = 1
up n x y = up (n-1) x (up n x (y-1))

File: 375/Easy.jl
# https://www.reddit.com/r/dailyprogrammer/comments/aphavc/20190211_challenge_375_easy_print_a_new_number_by/

function addOneToDigs(n)
  blog = floor(Int, log10(n))
  digs = (x -> x+1).([div(n % 10^(i+1), 10^i) for i in blog:-1:0])
  reduce((acc, x) -> acc * (x == 10 ? 100 : 10) + x, digs)
end
